src/components/Form.js

```diff
@@ -60,16 +64,16 @@ export default class Form extends Component {
   }
 
   onChange = (formData, options={validate: false}) => {
-    const liveValidate = this.props.liveValidate || options.validate;
-    const errors = liveValidate ? this.validate(formData) :
-                                  this.state.errors;
-    const errorSchema = toErrorSchema(errors);
-    setState(this, {
-      status: "editing",
-      formData,
-      errors,
-      errorSchema
-    }, () => {
+    const mustValidate = this.props.liveValidate || options.validate;
+    let state;
+    if (!mustValidate) {
+      state = {status: "editing", formData};
+    } else {
+      const errors = this.validate(formData);
+      const errorSchema = toErrorSchema(errors);
+      state = {status: "editing", formData, errors, errorSchema};
+    }
+    setState(this, state, () => {
```

Comment:
Wooops I thought you were commenting on https://github.com/mozilla-services/react-jsonschema-form/pull/163

This PR has now landed, so efforts are more needed on #163. Though if you feel like something is definitely wrong in here, please file an issue about it :)